Api/Services/EventService.cs

```diff
@@ -362,22 +363,69 @@ public async Task<Result> DeleteUserFromEventAsync(int id, User user)
 
 
         /// <summary>
-        /// Get future events the user is interested in
+        /// Get a list of events that the user might have interest in.
         /// </summary>
+        /// <param name="category">Value to filter the search results through</param>
+        /// <param name="dateFrom">date used as a bottom boundry for the search</param>
+        /// <param name="dateUntil">date used as a upper boundry for the search</param>
+        /// <param name="order">value by which the search result should be ordered</param>
         /// <param name="user">User whos intered event we go over</param>
         /// <param name="page">Page number to return.</param>
         /// <param name="perPage">Items per page.</param>
-        /// <returns>Paginated list of events in which user is interested</returns>
-        public async Task<Result<Pagination<EventSummaryVM>>> GetEventsInterestedInAsync(User user, int page, int perPage)
+        /// <returns>Paginated list of events in which user might be interested in.</returns>
+        public async Task<Result<Pagination<EventSummaryVM>>> GetUserEventsAsync(EventParticipationCategory category, DateTime? dateFrom, DateTime? dateUntil, SearchOrder order, User user, int page, int perPage)
         {
-            var query = context.Users
-                .Where(u => u.Id == user.Id)
-                .SelectMany(u => u.EventParticipations.Select(e => e.Event))
-                .Where(u => u.Time > DateTime.UtcNow)
-                .OrderBy(u => u.Time)
-                .ProjectTo<EventSummaryVM>(mapper.ConfigurationProvider);
+            IQueryable<Event> events = context.Events;
 
-            return await query.PaginateAsync(page, perPage, []);
+            switch (category)
+            {
+                case EventParticipationCategory.ParticipateIn:
+                    events = context.Users
+                        .Where(u => u.Id == user.Id)
+                        .SelectMany(u => u.EventParticipations
+                            .Where(e => e.DateDeleted == null && e.DateAccepted != null)
+                            .Select(e => e.Event));
+                    break;
+                case EventParticipationCategory.InterestedIn:
+                    events = context.Users
+                        .Where(u => u.Id == user.Id)
+                        .SelectMany(u => u.EventParticipations
+                            .Where(e => e.DateAccepted == null)
+                            .Select(e => e.Event));
+                    break;
+                case EventParticipationCategory.CreatedBy:
+                    events = events.Where(e => e.CreatorId == user.Id);
+                    break;
+            }
+
+            if (dateFrom != DateTime.MinValue)
+            {
+                events = events.Where(e => e.Time > dateFrom);
+            }
+
+            if (dateUntil != DateTime.MinValue)
+            {
+                events = events.Where(e => e.Time < dateUntil);
+            }
```

Comment:
Nie wydaje mi się, że to zadziała, nigdy nie przypisujesz takich wartości, lepiej sprawdzać czy nie są nullami